src/reducers/existingThoughtMove.ts

```diff
@@ -202,24 +258,27 @@ const existingThoughtMove = (state: State, { oldPath, newPath, offset }: {
       contextIndex: Index<Parent | null>,
       pendingMoves: { pathOld: Path, pathNew: Path }[],
     }
-    : Object.values(descendantUpdatesResult).reduce((accum, result) =>
+    : Object.values(lexemeUpdates).reduce((accum, result) =>
       result.contextsOld.reduce((accumInner, contextOld, i) => {
         const contextNew = result.contextsNew[i]
         const contextEncodedOld = hashContext(contextOld)
         const contextEncodedNew = hashContext(contextNew)
+
+        const childUpdate = childUpdates[hashContext(contextOld.concat(result.value))]
+
         const accumInnerChildrenOld = accumInner.contextIndex[contextEncodedOld]?.children
         const accumInnerChildrenNew = accumInner.contextIndex[contextEncodedNew]?.children
         const childrenOld = (accumInnerChildrenOld || getAllChildren(state, contextOld))
-          .filter((child: Child) => child.value !== result.value)
+          .filter((child: Child) => normalizeThought(child.value) !== normalizeThought(result.value))
```

Comment:
You can normalize `result.value` as soon as `result` is available (L261) for better performance. Otherwise it's called unnecessarily in both `result.contextsOld.reduce` and `.filter`.